src/components/ProductDetail/ProductDetail.js

```diff
@@ -88,6 +96,50 @@ class ProductDetail extends Component {
     this.selectVariant(variant);
   };
 
+  /**
+   * @name handleAddToCartClick
+   * @summary Called when the add to cart button is clicked
+   * @private
+   * @ignore
+   * @param {Number} quantity A positive integer from 0 to infinity, representing the quantity to add to cart
+   * @returns {undefined} No return
+   */
+  handleAddToCartClick = (quantity) => {
+    const {
+      currencyCode,
+      product,
+      uiStore: {
+        pdpSelectedOptionId,
+        pdpSelectedVariantId
+      }
+    } = this.props;
+
+    // Get selected variant or variant option
+    const selectedVariant = variantById(product.variants, pdpSelectedVariantId);
+    const selectedOption = variantById(selectedVariant.options, pdpSelectedOptionId);
+    const selectedVariantOrOption = selectedOption || selectedVariant;
+
+    if (selectedVariantOrOption) {
+      // Get the price for the currently selected variant or variant option
+      const price = priceByCurrencyCode(currencyCode, selectedVariantOrOption.pricing);
+
+      // Call addItemsToCart with an object matching the GraphQL `CartItemInput` schema
+      this.props.addItemsToCart([
```

Comment:
Should this be included in the destructure above instead of using `this.props` here?